modules/environment/aws/eks/main.tf

```diff
@@ -131,122 +135,98 @@ resource "aws_security_group" "elb" {
 }
 
 module "eks" {
-  source                                       = "terraform-aws-modules/eks/aws"
-  version                                      = "16.2.0"
-  cluster_version                              = var.cluster_version
-  cluster_name                                 = var.cluster
-  subnets                                      = sort(data.aws_subnet_ids.eks_masters.ids)
-  tags                                         = local.tags
-  vpc_id                                       = data.aws_vpc.lead_vpc.id
+  source  = "terraform-aws-modules/eks/aws"
+  version = "17.18.0"
+
+  cluster_name    = var.cluster
+  cluster_version = var.cluster_version
+  subnets         = sort(data.aws_subnet_ids.eks_masters.ids)
+  vpc_id          = data.aws_vpc.lead_vpc.id
+
   worker_additional_security_group_ids         = [aws_security_group.worker.id]
   map_roles                                    = concat(local.default_roles, local.codebuild_roles, var.additional_mapped_roles)
   write_kubeconfig                             = var.write_kubeconfig
   permissions_boundary                         = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/${aws_iam_policy.workspace_role_boundary.name}"
   manage_worker_iam_resources                  = true
   kubeconfig_aws_authenticator_additional_args = var.kubeconfig_aws_authenticator_additional_args
-  enable_irsa                                  = false
+  enable_irsa                                  = true
 
   cluster_endpoint_private_access                = true
   cluster_endpoint_public_access                 = var.enable_public_endpoint
   cluster_create_endpoint_private_access_sg_rule = true
-  cluster_endpoint_private_access_cidrs = [
-    "10.1.32.0/20",                  // internal VPN cidr
+  cluster_endpoint_private_access_cidrs          = [
+    "10.1.32.0/20", // internal VPN cidr
     data.aws_vpc.lead_vpc.cidr_block // anything running within the lead VPC, such as codebuild projects
   ]
+  cluster_enabled_log_types                      = ["api", "controllerManager", "scheduler"]
 
-  #cluster_enabled_log_types            = ["api","audit","authenticator","controllerManager","scheduler"]
+  tags = {
+    "Cluster" = var.cluster
+  }
 
-  workers_additional_policies = concat(["arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"], var.workers_additional_policies)
+  node_groups_defaults = {
+    create_launch_template = true
+    pre_userdata           = local.userdata
+    enable_monitoring      = true
+    key_name               = var.key_name
+    version                = var.cluster_version
+    disk_size              = var.root_volume_size
+    update_config          = {
+      max_unavailable_percentage = 50
+    }
 
-  workers_group_defaults = {
-    tags = [
-      {
-        "key"                 = "kubernetes.io/cluster-autoscaler/enabled"
```

Comment:
managed node groups automatically add these tags / labels by default, no need for them anymore